src/Models/BookModel.php

```diff
@@ -4,66 +4,149 @@
 
 use PDO;
 use Collection\Entities\Book;
+use Exception;
+use PDOException;
 
 class BookModel
 {
-  private PDO $db;
+    private PDO $db;
 
-  public function __construct(PDO $db)
-  {
-    $this->db = $db;
-  }
+    public function __construct(PDO $db)
+    {
+        $this->db = $db;
+    }
 
-  public function getAllBooks(): array | false
-  {
-    $query = $this->db->prepare("SELECT 
+    public function getAllBooks(): array | false
+    {
+        $query = $this->db->prepare("SELECT 
     `Books`.`id`, 
     `Books`.`title`, 
-    `Books`.`author`, 
+    `Books`.`author_id`, 
     `Books`.`isbn`, 
     `Books`.`pub_year`, 
     `Books`.`cover_img_url`, 
     `Books`.`summary`, 
     `Books`.`lang`, 
     `Books`.`gr_url`, 
-    `Books`.`notes`, 
     `Books`.`deleted`,
     `Authors`.`author_name`,
     `Authors`.`wiki_link`,
     `Languages`.`lang_name`
     FROM `Books`
-    INNER JOIN `Authors` ON `Books`.`author`=`Authors`.`id`
+    INNER JOIN `Authors` ON `Books`.`author_id`=`Authors`.`id`
     INNER JOIN `Languages` ON `Books`.`lang`=`Languages`.`id`;");
 
-    $query->execute();
+        $query->execute();
+
+        $data = $query->fetchAll();
+
+        if (!$data) {
+            return false;
+        }
+
+        $books = [];
+
+        foreach ($data as $book) {
+
+            $books[] = new Book(
+                $book['id'],
+                $book['title'],
+                $book['author_name'],
+                $book['wiki_link'],
+                $book['isbn'],
+                $book['pub_year'],
+                $book['cover_img_url'],
+                $book['summary'],
+                $book['lang_name'],
+                $book['gr_url'],
+                $book['deleted']
+
+            );
+        }
+        return $books;
+    }
+
+
+
+    public function getAllLanguages(): array
+    {
+        $query = $this->db->prepare("SELECT
+        `Languages`.`id`,
+        `Languages`.`lang_name`
+        FROM `Languages`
+        ");
+
+        $query->execute();
+
+        $languages = $query->fetchAll();
+
+        return $languages;
+    }
+
+    public function getAllAuthors(): array
+    {
+        $query = $this->db->prepare("SELECT
+        `Authors`.`id`,
+        `Authors`.`author_name`
+        FROM `Authors`
+        ");
 
-    $data = $query->fetchAll();
+        $query->execute();
 
-    if (!$data) {
-      return false;
+        $authors = $query->fetchAll();
+        return $authors;
     }
```

Comment:
Like how you have made a genre model, you sohuld also make a language and author model for these two methods.   
 

Purely because these two queries dont actually do anything in your books table. Joining other tables onto the books table is acceptable, but straight up querying other tables should be done in dedicated models

Reply:
done